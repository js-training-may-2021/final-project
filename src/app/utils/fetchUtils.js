import axios from 'axios';
import _ from 'lodash';

const isDev = process.env.NODE_ENV === 'development';
const pathApi = isDev ? 'http://localhost:3000' : '/api/v1';
export const limit = 20;

const fetchPokemons = (page, query, attr, order) => axios.get(`${pathApi}/pokemons?q=${query}&_page=${page}&_limit=${limit}&_sort=${attr}&_order=${order}`);
const normalizedQuery = (query) => _.trimStart(query, '#');

export const mappingFetchPokemons = {
  idAsc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'id', 'asc'),
  idDesc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'id', 'desc'),
  nameAsc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'name', 'asc'),
  nameDesc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'name', 'desc'),
};

export const fetchPokemonById = (id) => axios.get(`${pathApi}/pokemons?id=${id}`);

export const fetchData = () => (
  axios({
    method: 'post',
    url: '/api/v1/data',
    data: { token: localStorage.token },
    timeout: 4000,
  })
);

export const fetchCatch = (pokemon) => axios({
  method: 'post',
  url: '/api/v1/catchPokemon',
  data: { ...pokemon, token: localStorage.token },
  timeout: 4000,
});

export const fetchLogin = (data) => axios({
  method: 'post',
  url: '/api/v1/login',
  data,
  timeout: 4000,
});
