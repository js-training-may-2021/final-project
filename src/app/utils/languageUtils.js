import { useTranslation } from 'react-i18next';

export const setDefaultLanguage = () => {
  const defaultLanguage = 'ru';

  if (localStorage.getItem('language') === null) {
    localStorage.setItem('language', defaultLanguage);
  }
};

export const getCurrentLanguage = () => localStorage.getItem('language');

export const toggleLanguage = () => {
  const currentLanguage = localStorage.getItem('language');
  const newLanguage = currentLanguage === 'ru' ? 'en' : 'ru';
  localStorage.setItem('language', newLanguage);

  return newLanguage;
};

export const loc = (text) => {
  const { t } = useTranslation();
  return t(text);
};
