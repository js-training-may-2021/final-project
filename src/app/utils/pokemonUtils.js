import _ from 'lodash';

export const isFound = ({ name, id }, searchText) => {
  const searchBy = (attr, text) => String(attr).includes(text);
  const normalizedText = _.trimStart(searchText.trim().toLowerCase(), '#');
  return searchBy(name, normalizedText) || searchBy(id, normalizedText);
};

export const getNextId = (currentId) => {
  const nextId = currentId + 1 === 803 ? 10001 : currentId + 1;
  return nextId === 10148 ? 1 : nextId;
};

export const getPreviousId = (currentId) => {
  const prevId = currentId - 1 === 0 ? 10147 : currentId - 1;
  return prevId === 10000 ? 802 : prevId;
};

export const mappingSortPokemons = {
  idAsc: (array) => _.orderBy(array, 'id', 'asc'),
  idDesc: (array) => _.orderBy(array, 'id', 'desc'),
  nameAsc: (array) => _.orderBy(array, 'name', 'asc'),
  nameDesc: (array) => _.orderBy(array, 'name', 'desc'),
};
