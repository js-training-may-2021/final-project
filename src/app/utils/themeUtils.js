import { useSelector } from 'react-redux';

export const setDefaultTheme = () => {
  const defaultTheme = 'light';

  if (localStorage.getItem('theme') === null) {
    localStorage.setItem('theme', defaultTheme);
  }

  return localStorage.getItem('theme');
};

export const toggleTheme = () => {
  const currentTheme = localStorage.getItem('theme');
  const newTheme = currentTheme === 'light' ? 'dark' : 'light';
  localStorage.setItem('theme', newTheme);

  return newTheme;
};

export const isDarkTheme = () => {
  const currentTheme = useSelector((state) => state.uiState.theme);
  return currentTheme === 'dark';
};
