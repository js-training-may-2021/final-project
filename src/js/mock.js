export const pokemons = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: true,
    catchDate: '23 June 2021',
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmeleon',
    id: 5,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charizard',
    id: 6,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'squirtle',
    id: 7,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'wartortle',
    id: 8,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'blastoise',
    id: 9,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'caterpie',
    id: 10,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'metapod',
    id: 11,
    isCaught: false,
    catchDate: null,
  }];  

export const uncaughtPokemon = pokemons[0];
export const caughtPokemon = pokemons[1];
