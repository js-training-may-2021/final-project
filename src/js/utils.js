import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';

export const POKEMONS_PER_PAGE = 10;

let currentPage = 1;

export const getCurrentPage = () => {
	return currentPage++;
};

export const AppRoute = {
  MAIN: '/',
  DETAIL: '/detail/:id',
  CAUGHT: '/caught'
};

const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export const getCatchDate = () => {
  const date = new Date();

  const day = date.getDate();
  const month = monthNames[date.getMonth()];
  const year = date.getFullYear();

  return `${day} ${month} ${year}`
};

export const noop = () => {};
export const history = createMemoryHistory();
const mockStore = configureStore();
export const testStore = mockStore({});
