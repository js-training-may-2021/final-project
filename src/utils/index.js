export const firstLetterToUpperCase = (str) => {
  if (str === undefined) {
    return str;
  } else {
    return str
      .toLowerCase()
      .split(" ")
      .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
      .join(" ");
  }
};

export const formatDate = (dateInput) => {
  return dateInput.toLocaleString("ru-RU");
};

export const padToThree = (number) => number <= 999 ? `00${number}`.slice(-3) : number;

export const handleErrors = (response) => {
  if (!response.ok) {
    throw Error(`${response.status}: ${response.statusText}`);
  }
  return response;
};
