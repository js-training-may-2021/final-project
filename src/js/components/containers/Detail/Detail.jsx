import React, {Fragment, useEffect} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {useParams} from 'react-router';
import {getActivePokemon} from '../../../store/data/selectors.js';
import {Operation} from '../../../store/data/data.js';
import Header from '../../presentationals/Header/Header.jsx';
import Info from '../../presentationals/Info/Info.jsx';
import Loader from '../../presentationals/Loader/Loader.jsx';
import ErrorMessage from '../../presentationals/ErrorMessage/ErrorMessage.jsx';
import withToggle from '../../hocs/withToggle/withToggle.jsx';

const InfoWrapped = withToggle(Info);

export const Detail = (props) => {
  const {
    activePokemon,
		isLoading,
		errorMessage,
    onButtonClick,
		loadPokemon,
  } = props;

	
	const id = useParams().id;

	useEffect(() => {
		if (!activePokemon) {
			loadPokemon(id);
		}
	}, []);

	return (
		<Fragment>
			<Header />
			<main className="page__main page__main--detail">
				{activePokemon &&
					<InfoWrapped
						pokemon={activePokemon}
						isToggleChecked={activePokemon.isCaught}
						onButtonClick={onButtonClick}
					/>}
				{isLoading && <Loader />}
				{errorMessage && 
					<ErrorMessage
						message={errorMessage}
					/>}
			</main>
		</Fragment>
	);
};

Detail.propTypes = {
	activePokemon: PropTypes.shape({
		id: PropTypes.number.isRequired,
		name: PropTypes.string.isRequired,
		isCaught: PropTypes.bool.isRequired,
		catchDate: PropTypes.string,
	}),
	isLoading: PropTypes.bool.isRequired,
	errorMessage: PropTypes.string,
  onButtonClick: PropTypes.func.isRequired,
  loadPokemon: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
	activePokemon: getActivePokemon(state),
});

const mapDispatchToProps = dispatch => ({
	loadPokemon(id) {
    dispatch(Operation.loadPokemon(id));
	}
});

export default connect(mapStateToProps, mapDispatchToProps)(Detail);
