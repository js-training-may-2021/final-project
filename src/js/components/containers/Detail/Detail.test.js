import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import {Detail} from './Detail.jsx';
import {history, noop, testStore} from '../../../utils.js';
import {uncaughtPokemon, caughtPokemon} from '../../../mock.js';

it(`Render Detail page with uncaught pokemon`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={uncaughtPokemon}
            isLoading={false}
            errorMessage={null}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with caught pokemon`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={caughtPokemon}
            isLoading={false}
            errorMessage={null}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with Loader`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={null}
            isLoading={true}
            errorMessage={null}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with ErrorMessage`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={null}
            isLoading={false}
            errorMessage={'Failed to load pokemon. Please try again later.'}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with Loader and uncaught pokemon`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={uncaughtPokemon}
            isLoading={true}
            errorMessage={null}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with ErrorMessage and uncaught pokemon`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={uncaughtPokemon}
            isLoading={false}
            errorMessage={'Failed to load pokemon. Please try again later.'}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with Loader and caught pokemon`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={caughtPokemon}
            isLoading={true}
            errorMessage={null}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Detail page with ErrorMessage and caught pokemon`, () => {
  const tree = renderer
    .create(
      <Provider store={testStore}>
        <Router history={history}>
          <Detail
            activePokemon={caughtPokemon}
            isLoading={false}
            errorMessage={'Failed to load pokemon. Please try again later.'}
            onButtonClick={noop}
            loadPokemon={noop}
          />
          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
