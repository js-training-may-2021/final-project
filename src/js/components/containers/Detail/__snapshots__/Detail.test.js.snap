// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Detail page with ErrorMessage 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <p
      className="page__text page__text--message"
    >
      Failed to load pokemon. Please try again later.
    </p>
  </main>,
]
`;

exports[`Render Detail page with ErrorMessage and caught pokemon 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <img
      alt="Ivysaur"
      className="page__img"
      src="test-file-stub"
    />
    <article
      className="page__info info"
    >
      <h1
        className="info__title"
      >
        Ivysaur
      </h1>
      <p
        className="info__text"
      >
        ID: 
        <span
          className="info__data"
        >
          2
        </span>
      </p>
      <p
        className="info__text"
      >
        STATUS: 
        <span
          className="info__data"
        >
          Caught
        </span>
      </p>
      <p
        className="info__text"
      >
        CAUGHT: 
        <span
          className="info__data"
        >
          23 June 2021
        </span>
      </p>
      <button
        className="card__button button button--checked"
        onClick={[Function]}
        type="button"
      >
        Release
      </button>
    </article>
    <p
      className="page__text page__text--message"
    >
      Failed to load pokemon. Please try again later.
    </p>
  </main>,
]
`;

exports[`Render Detail page with ErrorMessage and uncaught pokemon 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <img
      alt="Bulbasaur"
      className="page__img"
      src="test-file-stub"
    />
    <article
      className="page__info info"
    >
      <h1
        className="info__title"
      >
        Bulbasaur
      </h1>
      <p
        className="info__text"
      >
        ID: 
        <span
          className="info__data"
        >
          1
        </span>
      </p>
      <p
        className="info__text"
      >
        STATUS: 
        <span
          className="info__data"
        >
          Not caught
        </span>
      </p>
      <button
        className="card__button button"
        onClick={[Function]}
        type="button"
      >
        Catch!
      </button>
    </article>
    <p
      className="page__text page__text--message"
    >
      Failed to load pokemon. Please try again later.
    </p>
  </main>,
]
`;

exports[`Render Detail page with Loader 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <p
      className="page__text page__text--message"
    >
      Loading...
    </p>
  </main>,
]
`;

exports[`Render Detail page with Loader and caught pokemon 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <img
      alt="Ivysaur"
      className="page__img"
      src="test-file-stub"
    />
    <article
      className="page__info info"
    >
      <h1
        className="info__title"
      >
        Ivysaur
      </h1>
      <p
        className="info__text"
      >
        ID: 
        <span
          className="info__data"
        >
          2
        </span>
      </p>
      <p
        className="info__text"
      >
        STATUS: 
        <span
          className="info__data"
        >
          Caught
        </span>
      </p>
      <p
        className="info__text"
      >
        CAUGHT: 
        <span
          className="info__data"
        >
          23 June 2021
        </span>
      </p>
      <button
        className="card__button button button--checked"
        onClick={[Function]}
        type="button"
      >
        Release
      </button>
    </article>
    <p
      className="page__text page__text--message"
    >
      Loading...
    </p>
  </main>,
]
`;

exports[`Render Detail page with Loader and uncaught pokemon 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <img
      alt="Bulbasaur"
      className="page__img"
      src="test-file-stub"
    />
    <article
      className="page__info info"
    >
      <h1
        className="info__title"
      >
        Bulbasaur
      </h1>
      <p
        className="info__text"
      >
        ID: 
        <span
          className="info__data"
        >
          1
        </span>
      </p>
      <p
        className="info__text"
      >
        STATUS: 
        <span
          className="info__data"
        >
          Not caught
        </span>
      </p>
      <button
        className="card__button button"
        onClick={[Function]}
        type="button"
      >
        Catch!
      </button>
    </article>
    <p
      className="page__text page__text--message"
    >
      Loading...
    </p>
  </main>,
]
`;

exports[`Render Detail page with caught pokemon 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <img
      alt="Ivysaur"
      className="page__img"
      src="test-file-stub"
    />
    <article
      className="page__info info"
    >
      <h1
        className="info__title"
      >
        Ivysaur
      </h1>
      <p
        className="info__text"
      >
        ID: 
        <span
          className="info__data"
        >
          2
        </span>
      </p>
      <p
        className="info__text"
      >
        STATUS: 
        <span
          className="info__data"
        >
          Caught
        </span>
      </p>
      <p
        className="info__text"
      >
        CAUGHT: 
        <span
          className="info__data"
        >
          23 June 2021
        </span>
      </p>
      <button
        className="card__button button button--checked"
        onClick={[Function]}
        type="button"
      >
        Release
      </button>
    </article>
  </main>,
]
`;

exports[`Render Detail page with uncaught pokemon 1`] = `
Array [
  <header
    className="page__header header"
  >
    <nav
      className="header__nav-wrap"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          POKEDEX
        </a>
      </div>
      <ul
        className="header__menu menu"
      >
        <li
          className="menu__item"
        >
          <a
            aria-current="page"
            className="menu__link menu__link--selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            ALL
          </a>
        </li>
        <li
          className="menu__item"
        >
          <a
            aria-current={null}
            className="menu__link"
            href="/caught"
            onClick={[Function]}
          >
            CAUGHT
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="page__main page__main--detail"
  >
    <img
      alt="Bulbasaur"
      className="page__img"
      src="test-file-stub"
    />
    <article
      className="page__info info"
    >
      <h1
        className="info__title"
      >
        Bulbasaur
      </h1>
      <p
        className="info__text"
      >
        ID: 
        <span
          className="info__data"
        >
          1
        </span>
      </p>
      <p
        className="info__text"
      >
        STATUS: 
        <span
          className="info__data"
        >
          Not caught
        </span>
      </p>
      <button
        className="card__button button"
        onClick={[Function]}
        type="button"
      >
        Catch!
      </button>
    </article>
  </main>,
]
`;
